File: javascript/binary/binary.js
function Binary(binaryInput) {
  var validInput = binaryInput.match(/^[1|0]+$/);
  this.binaryInput = validInput ? binaryInput : 0;
}

Binary.prototype.toDecimal = function() {
  var result = 0;
  var max = this.binaryInput.length - 1;
  for (var i = max; i >= 0; i--) {
    var digit = parseInt(this.binaryInput[i]);
    result += digit * (2 ** (max - i));
  }

  return result;
}

module.exports = Binary;

File: ruby/binary/binary.rb
class Binary
  def self.to_decimal input

    # validation
    raise ArgumentError.new if !input.match(/^[01]+$/)

    result = 0;
    max = input.length - 1;

    input.reverse.split('').each_with_index do |item, index|
      digit = item.to_i
      modI = max - index
      result += digit * (2 ** (max - modI));
    end

    return result;
  end
end
